cat-home/src/test/java/com/dianping/cat/report/task/TaskHelperTest.java

```diff
@@ -24,45 +24,170 @@
-		cal.set(Calendar.HOUR_OF_DAY, 0);
-		cal.set(Calendar.MINUTE, 0);
-		cal.set(Calendar.SECOND, 0);
-		cal.set(Calendar.MILLISECOND, 0);
-		Date yesterdayZero = TaskHelper.yesterdayZero(cal.getTime());
-		Assert.assertEquals(cal.getTimeInMillis(), yesterdayZero.getTime() + 3600L * 1000 * 24);
-	}
-
-	@Test
-	public void testTodayZero() {
-		Calendar cal = Calendar.getInstance();
-		cal.set(Calendar.DAY_OF_YEAR, 1);
-		cal.set(Calendar.HOUR_OF_DAY, 0);
-		cal.set(Calendar.MINUTE, 0);
-		cal.set(Calendar.SECOND, 0);
-		cal.set(Calendar.MILLISECOND, 1);
-		Date todayZero = TaskHelper.todayZero(cal.getTime());
-		Assert.assertEquals(cal.getTimeInMillis(), todayZero.getTime() + 1L);
-	}
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
 
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testJoinIntArrayChar() {
+        Assert.assertEquals("1,2", TaskHelper.join(new int[]{1, 2}, ','));
+    }
+
+    @Test
+    public void testYesterdayZero() {
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.DAY_OF_YEAR, 1);
+        cal.set(Calendar.HOUR_OF_DAY, 0);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        Date yesterdayZero = TaskHelper.yesterdayZero(cal.getTime());
+        Assert.assertEquals(cal.getTimeInMillis(),
+                yesterdayZero.getTime() + 3600L * 1000 * 24);
+    }
+
+    @Test
+    public void testYesterdayZero2() {
+        Assert.assertEquals(new Date(1562626800000L),
+                TaskHelper.yesterdayZero(new Date(1562769422000L)));
+    }
+
+    @Test
+    public void testTodayZero() {
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.DAY_OF_YEAR, 1);
+        cal.set(Calendar.HOUR_OF_DAY, 0);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 1);
+        Date todayZero = TaskHelper.todayZero(cal.getTime());
+        Assert.assertEquals(cal.getTimeInMillis(), todayZero.getTime() + 1L);
+    }
+
+    @PrepareForTest({TaskHelper.class})
+    @Test
+    public void testTodayZero2() throws Exception {
+        Date date = new Date(1562799600000L);
+        PowerMockito.whenNew(Date.class).withAnyArguments().thenReturn(date);
+        Assert.assertEquals(date,
+                TaskHelper.todayZero(null));
+    }
+
+    @Test
+    public void testTodayZero3() {
+        Assert.assertEquals(new Date(1562626800000L),
+                TaskHelper.todayZero(new Date(1562683022000L)));
+    }
+
+    @Test
+    public void testJoinIntArray() {
+        final int[] array = {1, -8, -7, -7, -7, -7, -7, -253_968, -7, -24, -24,
+                -2147475464, 4088, -2147483647, 9, 33505273, 1, -8};
+
+        Assert.assertNull(TaskHelper.join((int[]) null, '\u0000',
+                -1609, -1599));
```

Comment:
This is a check if the array is null. You can shorten the test to be fit into one line
ie `Assert.assertNull(TaskHelper.join((int[]) null, '1', 2, 3));`
Same for the assertNulls in the other join methods